bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -268,6 +279,15 @@ protected void addProvider(Provider<E> provider) {
         }
     }
 
+    /**
+     * Returns the key for a given element
+     *
+     * @param element
+     *            element
+     * @return key (must not be null)
+     */
+    protected abstract K getKey(E element);
```

Comment:
The Registry interface has method get by the key - public E get(K key), so registries have the notion of a key, also the expectation is - this key to be unique